File: Anushka-Pachauri/SlidingWindow/LongestSubstringWithoutRepeatingCharacters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] startIndex=new int[128];
        int start=0;
        int end=0;
        int maxLen=0;
        char[] s1=s.toCharArray();
        if(s1.length==0){
            return 0;
        }
        while(end<s1.length){
            start=Math.max(start,startIndex[s1[end]]);
            maxLen=Math.max(maxLen,end-start+1);
            startIndex[s1[end]]=end+1; 
            end++;
        }
        return maxLen;
    }
}

File: NikhilKumarGupta/Slidingwindow/Longest Substring Without Repeating Characters.cpp
class Solution {
    public:
        int lengthOfLongestSubstring(string s) {
            
            vector<int>mpp(256,-1);
            int l=0;
            int r=0;
            int maxlen=0;
            int n=s.size();
    
            while(r<n){
                if(mpp[s[r]]!=-1){
                    if(mpp[s[r]]>=l){
                        l=mpp[s[r]]+1;
                    }
                }
                int len=r-l+1;
                maxlen=max(maxlen,len);
                mpp[s[r]]=r;
                r++;
            }
            return maxlen;
    
        }
    };